pkg/envoy/ads/stream.go

```diff
@@ -48,7 +49,8 @@ func (s *Server) StreamAggregatedResources(server xds_discovery.AggregatedDiscov
 	// This is the Envoy proxy that just connected to the control plane.
 	// NOTE: This is step 1 of the registration. At this point we do not yet have context on the Pod.
 	//       Details on which Pod this Envoy is fronting will arrive via xDS in the NODE_ID string.
-	proxy := envoy.NewProxy(kind, uuid, si, utils.GetIPFromContext(server.Context()))
+	streamID := atomic.AddInt64(&s.streamCount, 1)
```

Comment:
maybe rename `streamCount`? It took me a moment to realize where there was no decrement on this value.  The function `GetConnectedProxyCount` doing a len(on connected proxies) to get the number of streams not using `streamCount`.

Reply:
Good call! Renaming to streamID, although if you have an alternative happy to use it